src/main/haskell/kore/src/Kore/ASTPrettyPrint.hs

```diff
@@ -776,3 +780,15 @@ instance (MetaOrObject level, PrettyPrint (variable level))
         writeOneFieldStruct flags "FunctionalStringLiteral" l
     prettyPrint flags (FunctionalCharLiteral l) =
         writeOneFieldStruct flags "FunctionalCharLiteral" l
+
+instance (MetaOrObject level, PrettyPrint (variable level))
+    => PrettyPrint (Predicate level variable)
+  where
+    prettyPrint flags pat =
+        prettyPrint flags (unsafeCoerce pat :: PureMLPattern level variable)
```

Comment:
You are absolutely right, I just failed to notice `unwrapPredicate` exists. The reason I used `unsafeCoerce` is because it's just a newtype wrapper so it should always be safe (Coercible/`coerce` would not work because the constructor is hidden).

I'll change to `unwrapPredicate`